.github/workflows/run-build.yaml

```diff
@@ -156,14 +143,50 @@ env:
   ENVDIR: env.d
 
 jobs:
+  # Use the GitHub API to get the ref that this workflow was called with so
+  # that we can checkout the nextstrain/.github repo and invoke other actions
+  # and scripts at the same ref.
+  #
+  # We are able to use `github.repository` and `github.run_id` to get the
+  # current run of the workflow because:
+  # "When a reusable workflow is triggered by a caller workflow, the
+  # github context is always associated with the caller workflow."¹
+  #
+  # This job is required as long as we are unable to detect the workflow ref
+  # from within the reusable workflow using a GitHub context variable.
+  # See:
+  #   - https://github.com/actions/toolkit/issues/1264
+  #   - https://github.com/actions/runner/issues/2417
+  #
+  # ¹ https://docs.github.com/en/actions/using-workflows/reusing-workflows
+  #
+  reusable-workflow-sha:
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ github.token }}
+    steps:
+      - id: sha
+        name: Get reusable workflow sha via API
+        run: |
+          sha="$(gh api repos/${{ github.repository }}/actions/runs/${{ github.run_id }} \
+            | jq --raw-output '
+                .referenced_workflows
+              | map(select(.path | contains("nextstrain/.github/.github/workflows/run-build.yaml")))
+              | .[0].sha')"
```

Comment:
Updated to the new composite action using Github's OIDC token in e4dcc9a685b170f563d4ba3c50e2d4a3695459d2

Reply:
Huh, when I tried running the workflow in seasonal-flu, the [returned repository](https://github.com/nextstrain/seasonal-flu/actions/runs/5214654029/jobs/9411193857#step:2:3) is the calling workflow's repo (seasonal-flu) instead of the reusable workflow's repo. The [returned sha](https://github.com/nextstrain/seasonal-flu/actions/runs/5214654029/jobs/9411193857#step:2:4) _does_ correctly points to the sha for the reusable workflow.

I'll investigate a little more, but worst case we just have to hard-code the repo name (`nextstrain/.github`) in the reusable workflow.

Reply:
OH. I had totally missed the parsing of the repo from the [`job_workflow_ref` in the original prototype](https://github.com/tsibley/github-actions-testing/blob/7fbf8f0e895c6e80f2722e3a08925a4e27536cc9/.github/workflows/oidc.yaml#L21) 🤦‍♀️

Reply:
Fixed in https://github.com/nextstrain/.github/pull/44/commits/d4ffa795c7105ddb417927c1e91bdfc67c3b1b5b